### ISSUE TITLE
Generate sources as post-build event

### ISSUE THREAD

**Issue by zarusz** [jburzynski/TypeGen#38]
Hello,

Thanks for this project. I recently started to use it, so far works great!

I have a question, is it possible to somehow configure the TS source generation to run after specific project is build?

So essentially what I would want is to have all my ts files generated when my WebAPI (ASP.NET Core 2.1) project is built. Ideally this should work when building the project on local VS as well in CI (`dotnet build ...`). I tried the CLI tool, but it said `Error		Package 'TypeGen.DotNetCli 2.0.0' has a package type 'DotnetTool' that is not supported by project 'xxx.xxx.WebAPI'.`

**Comment by jburzynski** [comment]
Hi!

`TypeGen.DotNetCli` is now a global .NET CLI tool. TypeGen doesn't have a per-project .NET CLI tool right now, because the current format of per-project tools is planned to be changed by Microsoft in the near future.

So right now one solution would be to install `TypeGen.DotNetCli` as a global .NET CLI tool and then add `dotnet-typegen generate` as a post-build action (or execute `dotnet-typegen generate` after running `dotnet build` in the console).

I think installing TypeGen as a global .NET CLI tool is currently probably the best way to add it as a post-build action.

Hope this helps!

**Comment by zarusz** [comment]
Thanks for the information. I actually installed the 2.0.1 `TypeGen.DotNetCli` today and managed to have a post-build event command added on m WebAPI project: 
`dotnet-typegen -v generate`

It works like a charm and now on every project build I get TS models regenerated +1

However, the Azure DevOps Build pipelines don't seem to support global `dotnet` plugins at the moment ([link](https://github.com/Microsoft/azure-pipelines-tasks/issues/8291)). Will have to work around that somehow...

Cheers!

**Comment by jburzynski** [comment]
Maybe you can try to use `TypeGen.DotNetCli` version 1.6.7 (it's on NuGet)? It's the last version where it's a per-project tool. It uses TypeGen 1.6.7, so doesn't have the newest features of 2.0.1, but if you're not using them, maybe you can try this (at least temporarily).

I've noticed that in the thread about Azure someone has pointed out a [workaround](https://github.com/dotnet/cli/issues/8368#issuecomment-424852996) - have you tried it? (you probably have, but I'm asking just in case)

**Comment by zarusz** [comment]
Yes, I saw the workaround and it didn't work for me initially. It turns out that since the post-build command is running on my `MyProject.WebAPI` I had to tell `dotnet` to install it inside my project directory like this: 

`dotnet.exe tool install --tool-path MyProject.WebAPI TypeGen.DotNetCli --version 2.0.1`

I now happily have the models generating when the project is build in Visual Studio as well in Azure Pipelines. 

Thanks for helping and your prompt responses!
